youtube_dl/extractor/adn.py

```diff
@@ -40,9 +41,55 @@ class ADNIE(InfoExtractor):
             'description': 'md5:2f7b5aa76edbc1a7a92cedcda8a528d5',
         }
     }
-    _BASE_URL = 'http://animedigitalnetwork.fr'
+    _BASE_URL = 'https://animedigitalnetwork.fr'
+    _LOGIN_URL = 'https://animedigitalnetwork.fr/connexion'
+    _NETRC_MACHINE = 'ADN'
     _RSA_KEY = (0xc35ae1e4356b65a73b551493da94b8cb443491c0aa092a357a5aee57ffc14dda85326f42d716e539a34542a0d3f363adf16c5ec222d713d5997194030ee2e4f0d1fb328c01a81cf6868c090d50de8e169c6b13d1675b9eeed1cbc51e1fffca9b38af07f37abd790924cd3bee59d0257cfda4fe5f3f0534877e21ce5821447d1b, 65537)
 
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+        login_page = self._download_webpage(
+            self._LOGIN_URL, None, 'Downloading login page')
+
+        def is_logged(webpage):
+            return 'task=user.logout' in webpage or 'view=logout' in webpage
+
+        # Already logged in
+        if is_logged(login_page):
+            return
+
+        form_data = self._form_hidden_inputs('login-form', login_page)
+        if not form_data.get('return'):
```

Comment:
this might change in the future, this is not necessary as you're checking for error in response page.